app/jni/jni/KeyEncapsulation.c

```diff
@@ -42,11 +42,12 @@ JNIEXPORT jobject JNICALL Java_com_example_liboqstestapp_liboqs_KeyEncapsulation
     if (cls == NULL) { fprintf(stderr, "\nCould not find class\n"); return NULL; }
 
     // Get the Method ID of the constructor
+    // jmethodID constructor_meth_id_ = (*env)->GetMethodID(env, cls, "<init>", "(Lcom/example/liboqstestapp/liboqs/KeyEncapsulation;)V");
     jmethodID constructor_meth_id_ = (*env)->GetMethodID(env, cls, "<init>", "(Lcom/example/liboqstestapp/liboqs/KeyEncapsulation;)V");
     if (NULL == constructor_meth_id_) { fprintf(stderr, "\nCould not initialize class\n"); return NULL; }
 
     // Call back constructor to allocate a new instance, with an int argument
-    jobject _nativeKED = (*env)->NewObject(env, cls, constructor_meth_id_);
+    jobject _nativeKED = (*env)->NewObject(env, cls, constructor_meth_id_, obj); // TODO: This line seems to be suspicous.
```

Comment:
According to stackoverflow we need to pass the super class as 4th parameter to get it working.  
https://stackoverflow.com/questions/25363027/jni-getmethodid-not-working-for-constructor-of-inner-class